### DISCUSSION TITLE
Lightening Strike Count, Time & Distance

### DISCUSSION THREAD

**Discussion by craigccfl** [briis/hass-weatherflow2mqtt#176] [Q&A]
Home Assistant Supervised, Version 2022.8.1.
Home Assistant MQTT  version 6.1.2
Tempest Weather Station.

Weatherflow2mqtt was working fine until the HA 2022.8.0 upgrade. Have not tried rolling it back to see if it will work again. Currently HA is not recording/registering lightening strikes,  count, distance and last strike time. SWFL is the lightening capital of the world, it's nice data to monitor. The Tempest app is working fine. All other Tempest sensors are working fine in HA. 

My thanks to the Devs of this project.  Anyone else experience this issue?

**Comment by GlennGoddard** [comment]
We just had a lighting storm pass thru a couple of days ago and can verify that lightning sensor is available.
Do you happen to have MQTT Explorer installed?  If you do, can you look at the traffic and see if it is at your MQTT broker and just not making it to HA.

**Reply by craigccfl** [reply]
Everything appears to set properly. Will wait for this afternoon/evening storm, watch mqtt-explorer 

{
  "name": "Tempest ST-000****36 Lightning Count",
  "unique_id": "ST-000****36-lightning_strike_count",
  "icon": "mdi:weather-lightning",
  "state_topic": "homeassistant/sensor/weatherflow2mqtt_ST-000***36/observation/state",
  "value_template": "{{ value_json.lightning_strike_count }}",
  "json_attributes_topic": "homeassistant/sensor/weatherflow2mqtt_ST-000***36/lightning_strike_count/attributes",
  "device": {
    "identifiers": [
      "weatherflow2mqtt_ST-000***36"
    ],
    "manufacturer": "WeatherFlow",
    "name": "Tempest ST-000***36",
    "model": "Tempest",
    "sw_version": "172",
    "via_device": "weatherflow2mqtt_HB-000***73"
  }
}

# History

{"name": "Tempest ST-000***36 Lightning Count", "unique_id": "ST-000***36-lightning_strike_count", "icon": "mdi:weather-lightning", "state_topic": "homeassistant/sensor/weatherflow2mqtt_ST-000***36/observation/state", "value_template": "{{ value_json.lightning_strike_count }}", "json_attributes_topic": "homeassistant/sensor/weatherflow2mqtt_ST-00084636/lightning_strike_count/attributes", "device": {"identifiers": ["weatherflow2mqtt_ST-00084636"], "manufacturer": "WeatherFlow", "name": "Tempest ST-000***36", "model": "Tempest", "sw_version": "172", "via_device": "weatherflow2mqtt_HB-000***73"}}

**Comment by phasperhoven** [comment]
Just to chip in: I installed a weather station last week and added Weatherflow2mqtt to HA 2022.8.3. Most of it is working great, but last night's thunderstorm did not register.

BTW: great that this add-on is available, thank you!

**Comment by craigccfl** [comment]
Moved my weaterflow2mqtt installation from the HA server to a Docker Container, still have the same issue. SWFL, the lightening capital of the world and getting no readings in HA. All other sensors are working as expected. If I find a fix, I'll post. Appreciate the dev's work

**Comment by GlennGoddard** [comment]
Try to monitor your UDP traffic directly.  The app may be reporting lighting by more that your unit, so don't be certain based on that.

**Comment by craigccfl** [comment]
Here's the Transmitted message for Lightning Strike:

Transmitted messages:
Tempest ST-000xxx36 Last Lightning Strike (sensor.tempest_st_000xxx36_last_lightning_strike)
MQTT discovery data:
Topic: homeassistant/sensor/weatherflow2mqtt_ST-000xxx36/lightning_strike_time/config
Payload
name: Tempest ST-000xxx36 Last Lightning Strike
unique_id: ST-000xxx36-lightning_strike_time
device_class: timestamp
icon: mdi:clock-outline
state_topic: homeassistant/sensor/weatherflow2mqtt_ST-000xxx36/lightning_strike_time/state
value_template: '{{ value_json.lightning_strike_time }}'
json_attributes_topic: >-
  homeassistant/sensor/weatherflow2mqtt_ST-000xxx36/lightning_strike_time/attributes
device:
  identifiers:
    - weatherflow2mqtt_ST-000xxx36
  manufacturer: WeatherFlow
  name: Tempest ST-000xxx36
  model: Tempest
  sw_version: '172'
  via_device: weatherflow2mqtt_HB-00068873
platform: mqtt
Subscribed topics:
homeassistant/sensor/weatherflow2mqtt_ST-00084636/lightning_strike_time/attributes
1 most recently received message(s)
Received 5:49:25 PM
QoS: 0, Retained
Payload: attribution: Powered by WeatherFlow2MQTT
homeassistant/sensor/weatherflow2mqtt_ST-000xxx36/lightning_strike_time/state
0 most recently received message(s)
Transmitted messages:
Tempest ST-000xxx36 Precipitation Type (sensor.tempest_st_000xxx36_precipitation_type)
MQTT discovery data:
Topic: homeassistant/sensor/weatherflow2mqtt_ST-000xxx36/precipitation_type/config
Payload

**Comment by craigccfl** [comment]
WeatherFlow to mqtt is transmitted, from mqtt to HA not registering in the Graphs and  Dashboard. Boggling because everything else with working perfectly. All UDP and TCP ports are open on my Lan, do I need to open a UDP WAN port?

**Reply by GlennGoddard** [reply]
I see the setup of the sensor but I don't see any lightning data.
